okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -754,6 +741,55 @@ private void fixupXOkapiToken(ModuleDescriptor md, MultiMap reqHeaders, MultiMap
     }
   }
 
+  private static void streamHandle(ProxyContext pc, ReadStream<Buffer> readStream,
+                                   WriteStream<Buffer> mainWriteStream,
+                                   List<HttpClientRequest> logWriteStreams) {
+    List<WriteStream<Buffer>> writeStreams = new LinkedList<>();
+    writeStreams.add(mainWriteStream);
+    for (WriteStream<Buffer> w : logWriteStreams) {
+      writeStreams.add(w);
+    }
+    pumpOneToMany(readStream, writeStreams);
+    readStream.exceptionHandler(e
+        -> pc.warn("streamHandle: content exception ", e));
+    readStream.resume();
+  }
+
+  private static void pumpOneToMany(ReadStream<Buffer> readStream,
+                                    List<WriteStream<Buffer>> writeStreams) {
+
+    readStream.handler(data -> {
+      AtomicInteger pend = new AtomicInteger();
```

Comment:
Using an AtomicInteger won't work in all cases:
1. The drainHandler may be called asynchronously at a later time. It may happen that the ProxyService missed the time period where the WriteStream queue was full and only sees the drainHandler call. 
2. A WriteStream implementation may have a high watermark for writeQueueFull() and a low watermark for calling drainHandler. Example: https://github.com/eclipse-vertx/vert.x/blob/3.9.1/src/main/java/io/vertx/core/file/impl/AsyncFileImpl.java#L70-L71 where high watermark is 128 * 1024 and low watermark is half of this = 64 * 1024. 

This algorithm for a pauseAndResume method avoids any counting and only uses the current full status:
```
for each WriteStream
  if not full
    set drainHandler to null
  else
    set drainHandler to pauseAndResume
    pause readStream
    exit this method.
// this point is only reached if all WriteStreams are not full
resume readStream
```
Call this method after data has been written to the WriteStreams.

Reply:
If you by "exit this method" mean return the for loop entirely.. Then the buffer that is passed in the readStream handler is "lost" for all writers not visited in for loop.